reference/configuration/security.rst

```diff
@@ -22,19 +22,376 @@ key in your application configuration.
     namespace and the related XSD schema is available at:
     ``http://symfony.com/schema/dic/services/services-1.0.xsd``
 
+Configuration
+-------------
+
+**Basic Options**:
+
+* `access_denied_url`_
+* `always_authenticate_before_granting`_
+* `erase_credentials`_
+* `hide_user_not_found`_
+* `session_fixation_strategy`_
+
+**Advanced Options**:
+
+* `access_control`_
+* `acl`_
+* `encoders`_
+* `firewalls`_
+* `providers`_
+* `role_hierarchy`_
```

Comment:
Isn't it a bit strange that the core featurs of the security component (access control, firewalls, providers & role hierarchy) are placed as "Advanced options"? And something very small (`access_denied_url`) is placed as nr 1 option to use?

I think an average application uses the second list of options more than the first list. As this is a reference, I don't think it nessarily as to focus on "beginners": that's the goal of the guides. This has to be a complete as possible list of options (and imo, ordered by frequent usage).

Reply:
Although you are right, the options are sorted alphabetically, which is what we do in the rest of the docs.

In any case, after this PR I intend to send a lot more related to Security because the actual situation is quite a mess. Things should be explained better. This reference is going to be much smaller and contain only links to articles where things are explained better.

For example, password hashing shouldn't be explained in a section of the security reference. The reference should contain a link to a main article explaining everything about hashing. Same for firewalls/authentication mechanisms, etc.